### PR TITLE
Add audit logs for scope management

### PR DESCRIPTION
**Pull request by emswbandara**
Depends on https://github.com/wso2/carbon-identity-framework/pull/3575

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthScopeEventPublisherProxy.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    private OAuthScopeEventPublisherProxy() {
+
+    }
+
+    public static OAuthScopeEventPublisherProxy getInstance() {
+
+        return proxy;
+    }
+
+    public void publishPreAddScope(int tenantId, ScopeDTO scope) {
```
Comment (mevan-karu) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthScopeEventPublisherProxy.java:48]: Better to add method comments to public methods.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/ScopeManagementAuditLogger.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        } else {
+            user = CarbonConstants.REGISTRY_SYSTEM_USERNAME;
+        }
+        return user;
+    }
+
+    private String buildScopeData(Event event) {
+
+        String name =
+                (String) event.getEventProperties().get("scopeName");
```
Comment (mevan-karu) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/ScopeManagementAuditLogger.java:106]: Can't we use the `IdentityEventConstants.EventProperty.SCOPE_NAME` in here? Check other places also.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/ScopeManagementAuditLogger.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        StringBuilder data = new StringBuilder();
+        data.append("Name:").append(name).append(", Description:").append(description).append(", Display Name:")
+                .append(displayName).append(", Claims:[");
+        if (ArrayUtils.isNotEmpty(claims)) {
+            String joiner = "";
+            for (String claim : claims) {
+                data.append(joiner);
+                joiner = ", ";
+                data.append(claim);
+            }
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/ScopeManagementAuditLogger.java:123]: How about using StringUtils.join
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#join-java.lang.Object:A-java.lang.String-

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/ScopeManagementAuditLogger.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    public void handleEvent(Event event) throws IdentityEventException {
+
+        int tenantId = (int) event.getEventProperties().get(IdentityEventConstants.EventProperty.TENANT_ID);
+        String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
+        if (log.isDebugEnabled()) {
+            log.debug(event.getEventName() + " event received to ScopeManagementAuditLogger for the " +
+                    "tenant: " + tenantDomain);
+        }
+
+        if (IdentityEventConstants.Event.POST_ADD_SCOPE.equals(event.getEventName())) {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/ScopeManagementAuditLogger.java:62]: a switch would make the code much readable

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/ScopeManagementAuditLogger.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+            String scopeName =
+                    (String) event.getEventProperties().get(IdentityEventConstants.EventProperty.SCOPE_NAME);
+            String initiator = getInitiatorUsername(tenantDomain);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/ScopeManagementAuditLogger.java:65]: These two lines seem to be redundant for all events. How about taking them out of the if block

**Comment by emswbandara** [comment]
closing until audit log format is finalized.

**Pull request closed without merging**